syntax = "proto3";

package cafm.identity.permission;

import "permission_context.proto";
import "order.proto";

message PermissionContextAppService_permissionContextByIdRequest {
  string id = 1;
}

message PermissionContextAppService_permissionContextByIdResponse {
  PermissionContext permissionContext = 1;
}

message PermissionContextAppService_permissionContextsRequest {
  int32 resultFrom = 1;
  int32 resultSize = 2;
  repeated order.Order order = 3;
}

message PermissionContextAppService_permissionContextsResponse {
  repeated PermissionContext permissionContexts = 1;
  int32 itemCount = 2;
}


service PermissionContextAppService {
  rpc permissionContextById (PermissionContextAppService_permissionContextByIdRequest) returns (PermissionContextAppService_permissionContextByIdResponse) {}
  rpc permissionContexts (PermissionContextAppService_permissionContextsRequest) returns (PermissionContextAppService_permissionContextsResponse) {}
}